File: src/main/java/pl/patrykdolata/lifemanager/config/WebConfigurer.java
package pl.patrykdolata.lifemanager.config;

import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import org.springframework.boot.web.servlet.ServletContextInitializer;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import org.springframework.core.env.Environment;
import org.springframework.util.CollectionUtils;
import org.springframework.web.cors.CorsConfiguration;
import org.springframework.web.cors.UrlBasedCorsConfigurationSource;
import org.springframework.web.filter.CorsFilter;

import javax.servlet.ServletContext;
import javax.servlet.ServletException;

@Configuration
public class WebConfigurer implements ServletContextInitializer {

    private final Logger log = LoggerFactory.getLogger(WebConfigurer.class);

    private final Environment environment;
    private final SecurityProperties securityProperties;

    public WebConfigurer(Environment environment, SecurityProperties securityProperties) {
        this.environment = environment;
        this.securityProperties = securityProperties;
    }

    @Override
    public void onStartup(ServletContext servletContext) throws ServletException {
        if (environment.getActiveProfiles().length != 0) {
            log.info("Web application configuration, using profiles: {}", (Object[]) environment.getActiveProfiles());
        }
        log.info("Web application fully configured");
    }

    @Bean
    public CorsFilter corsFilter() {
        UrlBasedCorsConfigurationSource source = new UrlBasedCorsConfigurationSource();
        CorsConfiguration corsConfiguration = securityProperties.getCors();
        if (!CollectionUtils.isEmpty(corsConfiguration.getAllowedOrigins())) {
            log.debug("Registering CORS filter");
            source.registerCorsConfiguration("/api/**", corsConfiguration);
            source.registerCorsConfiguration("/v2/api-docs", corsConfiguration);
        }
        return new CorsFilter(source);
    }
}

File: src/main/kotlin/pl/patrykdolata/lifemanager/config/WebConfigurer.kt
package pl.patrykdolata.lifemanager.config

import org.slf4j.Logger
import org.slf4j.LoggerFactory
import org.springframework.boot.web.servlet.ServletContextInitializer
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.core.env.Environment
import org.springframework.web.cors.UrlBasedCorsConfigurationSource
import org.springframework.web.filter.CorsFilter
import javax.servlet.ServletContext

@Configuration
class WebConfigurer(private val environment: Environment, private val securityProperties: SecurityProperties)
    : ServletContextInitializer {

    private val log: Logger = LoggerFactory.getLogger(WebConfigurer::class.java)

    override fun onStartup(servletContext: ServletContext?) {
        if (environment.activeProfiles.isNotEmpty()) {
            log.info("Web application configuration, using profiles: ${environment.activeProfiles}")
        }
        log.info("Web application fully configured")
    }

    @Bean
    fun corsFilter(): CorsFilter {
        val source = UrlBasedCorsConfigurationSource()
        val corsConfiguration = securityProperties.cors
        if (!corsConfiguration.allowedOrigins.isNullOrEmpty()) {
            log.debug("Registering CORS filter")
            source.registerCorsConfiguration("/api/**", corsConfiguration)
            source.registerCorsConfiguration("/v2/api-docs", corsConfiguration)
        }
        return CorsFilter(source)
    }
}
